src/CharterManager.sol

```diff
@@ -278,13 +278,15 @@ contract CharterManagerImp {
         _validate(ilk, u, urp, dink, dart, rate, spot, _gate);
     }
 
-    function flux(bytes32 ilk, address src, address dst, uint256 wad) external allowed(src) {
+    function flux(address gemJoin, address src, address dst, uint256 wad) external allowed(src) {
         address surp = getOrCreateProxy(src);
```

Comment:
This needs `require(VatLike(vat).wards(crop) == 1, "ChaterManager/crop-not-authorized")`

Reply:
I don't think we need it as we don't do an external call to that address.
So it's similar to how we don't have it on frob now.

I do agree that we need to either not have it on both charter and crop join, or have it on both.

Reply:
Well actually we might not need to have it the same in both, as in crop join we do have the external call.
So I still think we can skip it here, as we could of just passed `ilk`, which wouldn't have any such sanity check.

Reply:
We do an external call to `ilk()`. Granted it's a view, so nothing can be modified.